x-pack/plugins/profiling/server/routes/setup/route.ts

```diff
@@ -90,45 +74,59 @@ export function registerSetupRoute({
           useDefaultAuth: false,
         });
 
-        const commonParams = {
+        const commonSetupParams: ProfilingSetupOptions = {
           client: clientWithDefaultAuth,
+          clientWithProfilingAuth,
           logger,
-          packagePolicyClient: dependencies.start.fleet.packagePolicyService,
           soClient: core.savedObjects.client,
           spaceId:
             dependencies.setup.spaces?.spacesService?.getSpaceId(request) ?? DEFAULT_SPACE_ID,
-          isCloudEnabled,
         };
 
-        const setupState = await dependencies.start.profilingDataAccess.services.getSetupState(
-          commonParams,
-          clientWithProfilingAuth
-        );
-
-        const executeAdminFunctions = [
-          ...(setupState.resource_management.enabled ? [] : [enableResourceManagement]),
-          ...(setupState.permissions.configured ? [] : [setSecurityRole]),
-          ...(setupState.settings.configured ? [] : [setMaximumBuckets]),
-        ];
-
-        const executeViewerFunctions = [
-          ...(setupState.policies.collector.installed ? [] : [createCollectorPackagePolicy]),
-          ...(setupState.policies.symbolizer.installed ? [] : [createSymbolizerPackagePolicy]),
-          ...(setupState.policies.apm.profilingEnabled
-            ? [removeProfilingFromApmPackagePolicy]
-            : []),
-        ];
-
-        if (!executeAdminFunctions.length && !executeViewerFunctions.length) {
-          return response.ok();
+        const { type, setupState } =
+          await dependencies.start.profilingDataAccess.services.getSetupState({
+            esClient,
+            soClient: core.savedObjects.client,
+            spaceId:
+              dependencies.setup.spaces?.spacesService?.getSpaceId(request) ?? DEFAULT_SPACE_ID,
+          });
+
+        const isCloudEnabled = dependencies.setup.cloud?.isCloudEnabled;
+        if (isCloudEnabled && type === 'cloud') {
+          if (!dependencies.start.fleet) {
+            const msg = `Elastic Fleet is required to set up Universal Profiling on Cloud`;
+            logger.error(msg);
+            return response.custom({
+              statusCode: 500,
+              body: { message: msg },
+            });
+          }
+          logger.debug('Setting up Universal Profiling on Cloud');
+
+          await setupCloud({
+            setupState,
+            setupParams: {
+              ...commonSetupParams,
+              packagePolicyClient: dependencies.start.fleet.packagePolicyService,
+              isCloudEnabled,
+              config: dependencies.config,
+            },
+          });
+
+          logger.debug('[DONE] Setting up Universal Profiling on Cloud');
+        } else {
```

Comment:
@cauemarcondes, for version 8.12, we aim to limit profiling to a select group of customers on self-managed platforms in order to gather initial feedback. To accomplish this, we need to introduce a manual setting, similar to how we used `xpack.profiling.enabled` during the beta phase for cloud deployments. I'd suggest `xpack.profiling.selfmanaged.enabled` for this here. FYI @inge4pres @sboomsma 

In summary, self-managed will be in beta for 8.12. Once we are comfortable with it, we will then enable it by default in a future release

Reply:
sounds good to me @iogbole ... I'll make the changes...

Reply:
After talking to @iogbole we decided to keep using the current setting (`xpack.profiling.enabled`) only. This won't change this current behaviour in the Cloud as Profiling is enabled by default, and for testing the new on-prem flow users would need to enable the setting manually. Internally the Profiling set up logic knows when on cloud or on-prem to set it up accordingly.

Reply:
Thanks for the write up 🙏🏼 
Starting to test this now